libfoundation/src/foundation-unicode.cpp

```diff
@@ -378,7 +378,7 @@ bool MCUnicodeGetProperty(const unichar_t *p_chars, uindex_t p_char_count, MCUni
             {
                 int32_t t_prop;
                 t_prop = MCUnicodeGetIntegerProperty(t_char, p_property);
-                if (t_prop < UINT8_MIN || t_prop > UINT8_MAX)
+                if (t_prop < 0 || t_prop > UINT8_MAX)
                     return false;
```

Comment:
That fixes the issue with `-1 < 0u` returning false - but more broadly, all the UINT*_MIN could be changed from `0u` to `0`, to avoid the operands cast like here.